cms/server/templates/contest/task_submissions.html

```diff
@@ -101,21 +101,41 @@ <h2 style="margin-bottom: 10px">{{ _("Submit a solution") }}</h2>
 {% end %}
 
 <div id="submit_solution" class="row">
-    <div class="span5">
+    <div class="span9">
         <form class="form-horizontal" enctype="multipart/form-data" action="{{ url_root }}/tasks/{{ encode_for_url(task.name) }}/submit" method="POST">
-            <fieldset>
+            <fieldset class="row-fluid">
 {% for idx, filename in enumerate(x.filename for x in task.submission_format) %}
                 <div class="control-group">
-                    <label class="control-label" for="input{{ idx }}">{{ filename.replace(".%l", "") }}</label>
-                    <div class="controls">
+                        <label class="control-label span2" for="input{{ idx }}">{{ filename.replace(".%l", "") }}</label>
+
+                    <div class="span4">
                         <input type="file" class="input-xlarge" id="input{{ idx }}" name="{{ filename }}"/>
+                        <!-- FIXME: figure out why we had div class="controls" wrapping this -->
+                    </div>
+
+                    <!-- FIXME: What does this even do when there are multiple inputs? -->
```

Comment:
Oh, when I wrote that I was thinking about what happens if we have to submit more than one file (not a single zip of multiple files). I didn't think we'd ever have to do that for anything, let alone for the AIO so I added a note and left it at that. But yeah, as you said, user tests does exactly that :p (I think). 

I'll have a closer look at UserTestHandler at some point, but it looks like it'll be messy since you can submit tests as a zip or as separate files.